src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -157,57 +157,74 @@ private function parseDefinitions($content, $file)
         if (!is_array($content['services'])) {
             throw new InvalidArgumentException(sprintf('The "services" key should contain an array in %s. Check your YAML syntax.', $file));
         }
-        if (isset($content['services']['_defaults'])) {
-            if (!is_array($defaults = $content['services']['_defaults'])) {
-                throw new InvalidArgumentException(sprintf('Service defaults must be an array, "%s" given in "%s".', gettype($defaults), $file));
+
+        list($content, $defaults) = $this->parseDefaults($content, $file);
+        foreach ($content['services'] as $id => $service) {
+            $this->parseDefinition($id, $service, $file, $defaults);
+        }
+    }
+
+    /**
+     * Parses the "_defaults" key.
+     *
+     * @param array  $content
+     * @param string $file
+     *
+     * @return array
+     *
+     * @throws InvalidArgumentException
+     */
+    private function parseDefaults($content, $file)
```

Comment:
I don't think the docblock gives any value here, we could remove it.
Missing `array` type hint :)
Shouldn't $content be passed by ref, and the return value just be `$default` instead of `array($content, $defaults)`?